Newsletters: add 175 (2021-11-17)

harding opened the pull request:
- [x] Add topics and update releases/RCs @harding 
- [x] Client/services section @bitschmidty 
- [x] `Bitcoin Core #23077` Full CJDNS support @Xekyo 
- [x] `Eclair #1957` Add basic support for onion messages `lightning/bolts#759` @dongcarl

Xekyo reviewed (commented):
LGTM so far. Will work on my write-up today.

jnewbery commented on _posts/en/newsletters/2021-11-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[BDK][bdk repo], [Bitcoin Improvement Proposals (BIPs)][bips repo], and
+[Lightning BOLTs][bolts repo].*
```
[bolts repo] should be updated to `https://github.com/lightning/bolts` (see https://lists.linuxfoundation.org/pipermail/lightning-dev/2021-November/003337.html). Perhaps this should also be a short news item as well?

harding replied:
This link is defined in `_includes/references.md` and updating it will update the links in all previous newsletters.  I think that's fine, but I don't want to do it at the last minute before the publication of this newsletter, so I'll do it in a separate PR.  We can also use that separate PR to update some related links.

Regarding a news item, I thought about it a week or two ago when it happened but decided it was mainly administrivia not development news.  However, if you or anyone else has strong feeling about it, I'm happy to add an item to next week's newsletter about it.

jnewbery replied:
I agree. Thanks Dave.

jnewbery commented on _posts/en/newsletters/2021-11-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+  to [BIP340][] (see [Newsletter #87][news87 bips886]) previously discussed in
+  [Newsletter #83][news83 safety].
+
+- [Bitcoin Core #23077][] enables address relay via CJDNS, making CJDNS
```
Perhaps link to https://github.com/cjdelisle/cjdns for readers who aren't familiar with CJDNS.

jnewbery commented on _posts/en/newsletters/2021-11-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+  interpreted as private IPv6 addresses.
+
+<!-- FIXME: harding to add topic for onion messages -->
+- [Eclair #1957][] adds basic support for [onion messages][news92 onion
+  messages] per [BOLTs #759][], allowing relaying of onion messages, but
```
Would it be better to link to http://localhost:4000/en/newsletters/2020/02/26/#ln-direct-messages for onion messages (which is the description of the proposal, rather than the c-lightning implementation)?

bitschmidty replied:
This is resolved by the new topic addition for onion messages.

jnewbery reviewed (commented):
Sorry for late review. Looks great. Just a few minor comments.

harding commented on _posts/en/newsletters/2021-11-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+
+- **Muun wallet supports taproot:**
+  Muun wallet enabled taproot address support after activation occurred,
+  including the ability for users to default to taproot receive addresses.
```
@bitschmidty Is it deliberate that there's no link for the Muun item?  It's fine if deliberate (e.g. there's no good link).

bitschmidty replied:
Yes, there wasn’t a good link but lots of twitter screenshots only.

harding commented on _posts/en/newsletters/2021-11-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+
+- **Lightning-based social feed squeaknode launches:**
+  [Squeaknode][squeaknode github] is a Twitter-like P2P network based on LN
+  (using LND). Users can create, view, buy, and sell messages known as "squeaks".
```
I don't really like the Squeaknode item.  I don't think we want to announce every new LN-using service.  However, I don't have any particular strong objection to this particular instance.

bitschmidty replied:
Point taken. Id like to leave it in as its an open source project, p2p, with its own protocol and seems to have some traction (being included in umbrel). But I do understand that we cannot be an application layer LN monthly digest for this section.

jnewbery replied:
I was also leaning against including squeaknode, and hesitated before not leaving a comment. I think that we should reserve this section to highlight products that are using Bitcoin or Lightning protocol features. There are other publications that cover lightning apps.

It sounds like Dave and I are both weak NACKs, so if you're a strong ACK @bitschmidty, we can leave it in.

bitschmidty replied:
Im not that strong on it. Removing.

harding commented on _posts/en/newsletters/2021-11-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+- **Kollider launches alpha LN-based trading platform:**
+  Kollider's latest [announcement][kollider blog] details the derivative
+  platform's features including LN deposits and withdrawals, LNAUTH and LNURL
+  support, and a LN-based referral scheme.
```
I'm going to remove the part about an LN-based referral scheme. Referral schemes are kinda sleezy feeling[*] and I don't think mentioning it here really adds anything (it's just a trusted payout via LN for referrals, not some new fancy trustless scheme, AFAICT).

[*] Says the guy whose first full time job as a technical writer was for a company that made referral tracking software.

bitschmidty replied:
ACK

harding reviewed (commented):
@bitschmidty

harding commented:
Added a new topic for onion messages; added topic entries; updated RCs; reviewed work from the other contributors (thanks all!), and made edits for @jnewbery feedback (thanks!).

jnewbery commented on _topics/en/onion-messages.md:
```diff
@@ -0,0 +1,54 @@
+  by nodes that support the protocol.  Messages don't use HTLCs,
+  minimizing the use of LN node resources.
+---
+
```
(Can be done in a follow-up PR). Perhaps add a note that this topic refers to a specific feature in LN, and not onion routing or tor (similar to the note in the transaction bloom filtering topic: "_Note: this topic only refers to BIP37 transaction bloom filters.  There are uses of generic bloom filters in Bitcoin (such as in Bitcoin Core's transaction relay tracking) that aren't indexed to this topic._")

jnewbery reviewed (commented):
All looks good to me. Thanks team!

bitschmidty commented:
Good work team! @harding @Xekyo @jnewbery thank you!
